modules/cms/widgets/MediaManager.php

```diff
@@ -985,10 +985,14 @@ protected function checkUploadPostback()
 
             $path = Input::get('path');
             $path = MediaLibrary::validatePath($path);
-
+            
+            $realpath = empty(trim($uploadedFile->getRealPath())) 
+                            ? $uploadedFile->getPath().DIRECTORY_SEPARATOR.$uploadedFile->getFileName() 
+                            : $uploadedFile->getRealPath()
+                        ;
             MediaLibrary::instance()->put(
                 $path.'/'.$fileName,
-                File::get($uploadedFile->getRealPath())
+                File::get($realpath))
```

Comment:
Also why is there a need for this glued together version of realpath?
And `$realpath` should be `$realPath` by the coding standards: http://octobercms.com/help/guidelines/developer#variable-naming